src/Vipps.net/VippsApi.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Net.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Vipps.net.Infrastructure;
+using Vipps.net.Services;
+
+namespace Vipps.net
+{
+    public interface IVippsApi
+    {
+        IVippsAccessTokenService AccessTokenService();
+        IVippsEpaymentService EpaymentService();
+        IVippsCheckoutService CheckoutService();
+    }
+
+    public class VippsApi : IVippsApi
+    {
+        private readonly VippsConfigurationOptions _vippsConfigurationOptions;
+        private readonly VippsHttpClient _vippsHttpClient;
+        private readonly VippsAccessTokenService _accessTokenService;
+        private readonly IVippsEpaymentService _epaymentService;
+        private readonly IVippsCheckoutService _checkoutService;
+        private readonly ILoggerFactory _loggerFactory;
+
+        public VippsApi(
+            VippsConfigurationOptions configurationOptions,
+            HttpClient httpClient = null,
+            ILoggerFactory loggerFactory = null
+        )
+        {
+            _loggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
+            _vippsConfigurationOptions = configurationOptions;
+            _vippsHttpClient = new VippsHttpClient(httpClient, configurationOptions);
+
+            _accessTokenService = new VippsAccessTokenService(
+                configurationOptions,
+                new AccessTokenServiceClient(
+                    configurationOptions,
+                    _vippsHttpClient,
+                    _loggerFactory
+                ),
+                new AccessTokenCacheService()
+            );
+
+            _epaymentService = new VippsEpaymentService(
+                new EpaymentServiceClient(
+                    _vippsConfigurationOptions,
+                    _vippsHttpClient,
+                    _accessTokenService,
+                    _loggerFactory
+                )
+            );
+
+            _checkoutService = new VippsCheckoutService(
+                new CheckoutServiceClient(
+                    _vippsConfigurationOptions,
+                    _vippsHttpClient,
+                    _loggerFactory
+                )
+            );
+        }
+
+        public IVippsAccessTokenService AccessTokenService()
+        {
+            return _accessTokenService;
+        }
+
+        public IVippsEpaymentService EpaymentService()
+        {
+            return _epaymentService;
+        }
+
+        public IVippsCheckoutService CheckoutService()
```

Comment:
Does this need to be a function? can the constructor do this instead?

Reply:
I was thinking making them properties instead, if that's what you mean? 
There's no logic here, only reading a private field

Reply:
I misunderstood a bit. i see it is done in the constructor. I rephrase the question to: why does this need to be a callable function? cant the checkoutservice just be accessed as a public member of the class?

Reply:
Yeah agreed

Reply:
Fixed